ddprof-lib/src/main/cpp/symbols_linux.cpp

```diff
@@ -418,8 +427,74 @@ void Symbols::parseKernelSymbols(CodeCache* cc) {
     fclose(f);
 }
 
+namespace {
```

Comment:
Talking with @nsavoire about the approach:
- It would be better to rely on an equivalent of `process_vm_readv`.
That would avoid the problems associated with segv handlers.
- If we go for the segvhandler solution
there are helpers available within async-profiler to do this in a safer way.
- The current solution only reduces the risk of a race condition (as we can still have a mapping unloaded between time of check and time of use)

Sources that relate to this:
- https://github.com/abseil/abseil-cpp/blob/3848ed7f1b13259a8ddd35dc56bf216ecd36d968/absl/debugging/internal/address_is_readable.cc#L52
- https://github.com/async-profiler/async-profiler/blob/179b0b1285c656bb7afae3f144d540256547b522/src/profiler.cpp#L835

Reply:
> The current solution only reduces the risk of a race condition (as we can still have a mapping unloaded between time of check and time of use)

Indeed. We would need to use `SafeAccess:load*` calls for all memory loads in the elf parser if the mapping can just disappear in the middle of processing :(

Reply:
I am assuming the handlers are not setup at that time (when we are doing the parse library function).
I'll see if I can make a version using SafeAcess, though I am not too familiar with profiler init sequences as well as the whole SafeAccess infrastructure.
The important part to protect is the access to the section header, which is the first read into the mapping.
```
_header->e_shoff
```